node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageSender.scala

```diff
@@ -57,17 +57,13 @@ class PeerMessageSender(client: Client)(implicit np: NetworkParameters)
 
   private[node] def sendMsg(msg: NetworkPayload): Unit = {
     logger.debug(s"Sending msg=${msg.commandName} to peer=${socket}")
-    val newtworkMsg = NetworkMessage(np, msg)
+    val newtworkMsg = NetworkMessage(conf.network, msg)
     client.actor ! newtworkMsg
   }
 }
 
 object PeerMessageSender {
 
-  private case class PeerMessageSenderImpl(client: Client)(
```

Comment:
I noticed a few instances of `impl` being removed. Are we just replacing those with `serializers`? Is that the preferred coding practice?

Reply:
This is pattern happens a bunch of places in our code base:
- declare a trait
- make a companion object
- create a private case class that implemented the trait, and return this case class from the companion object methods.

Sometimes this is a necessary thing, but a bunch of times this isn't. So I'm trying to move away from this where we don't need it, as it is very verbose and causes more allocation of objects.